### DISCUSSION TITLE
Disable transition?

### DISCUSSION THREAD

**Discussion by bond2k** [helgeerbe/picframe#180] [Q&A] [answered]
Hi

I'm using a Raspberry Pi 4 to show images on a 4K tv. For some reason even a fade_time setting of 0.5 still takes a couple of seconds to transition from one image to another. Now I want to disable it, but I don't see an option to do that. Is it possible somehow?

Thanks

**Comment by jgodfrey** [comment]
[Looking at the code](https://github.com/helgeerbe/picframe/blob/ecc74ebc07efaf38994abd8b098c9b465fdb1bd5/picframe/viewer_display.py#L427), the minimum value of fade_time is 0.5 seconds. More correctly, it'll be forced to 0.5 seconds if you set it to something less...

I guess the real question is why the fade is taking so long to begin with? I know @sapnho is using a 4K monitor. Perhaps he has some input?

**Comment by sapnho** [comment]
Yes, indeed. There is a limit to the minimum speed that a Pi 4 can do on a 4K screen. While I personally love the effect of 10 seconds transitions, if you want to disable it, you will have to move to [feh](https://www.thedigitalpictureframe.com/the-best-3-image-viewers-for-your-raspberry-pi-digital-picture-frame-project/#feh).

**Reply by bond2k** [reply]
I tried with feh, but there i have other issues like "ERROR: Can't open X display". That's why I use this one. :)

**Reply by paddywwoof** [reply]
I will try to fix single frame transition tomorrow if I get a chance to look at it.

**Comment by paddywwoof** [comment]
Picframe uses OpenGL to allow interesting transitions between images but this means there is a finite time required to load each image from disk then convert it to a texture sampler. If the images on disk are different dimension from the screen, or the matting or picture pairs are used then there is an additional time for resizing with the python image library. However, all that dead time is supposed to take place before the transition starts so it shouldn't impact how quickly one slide fades to another, it limits the minimum display time for each slide.

For the fading from one image to the next there is a second limitation of using the GPU which is the fact that the frames per second is limited by the number of pixels being re-drawn and I have found that a full 4K screen can't go faster than 15FPS. This results is visible "jumps" during the fade, made worse because there are different time delays caused by background processes so the interval between frames varies (this happens if the transition time is set lower than the GPU can cope with, i.e. it is hardware limited).

But, obviously, there is no reason that the transition can't happen in a single frame apart from the fact that it wasn't designed to do that. The minimum `fade_time` was put in to prevent some secondary effect of trying to work out the alpha change from frame to frame during the fade. I'm not sure what would be involved but it probably wouldn't be too onerous to make a `fade_time` of 0.0 do the fade in a single frame (NB not looked at that bit of code for quite a while so rather indistinct idea of what might be involved)

**Comment by paddywwoof** [comment, marked as answer]
I've put a small change in the dev branch https://github.com/helgeerbe/picframe/commit/3bd1fb73c97ef735727b5b9ee63881e2a39095da that should allow single frame transitions. There is an issue if you have the text show time quite short as there is no 'buffer' at the start by adding the fade time (i.e. if fade time is 0) so the text showing time is eaten into by the file loading etc. If you have a short time showing text and long time loading images then then the text might have already faded away by the time it starts to be shown!

**Reply by bond2k** [reply]
Thanks a lot. I will try it, but first I need to figure out how to install it from that branch as I was just following the "normal" RaspPi installation. :)

**Comment by jgodfrey** [comment]
There are some instructions for installing (and running from) the `dev` branch here:

https://github.com/helgeerbe/picframe/issues/181#issuecomment-951193717

**Reply by bond2k** [reply]
worked perfectly, thanks
